internal/common/predicate.go

```diff
@@ -12,3 +14,14 @@ func IsControlledBy(object client.Object, owner *metav1.OwnerReference) bool {
 		controller.APIVersion == owner.APIVersion &&
 		controller.Kind == owner.Kind
 }
+
+// FetchCurrentNamespaceName returns namespace name by using information stored as file
+// Returns default Openshift Builds namespace on error
+// Refer: https://kubernetes.io/docs/tasks/run-application/access-api-from-pod/#without-using-a-proxy
+func FetchCurrentNamespaceName() string {
+	content, err := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
```

Comment:
An area for future improvement - create an "inner" function that captures the rest of the logic, but accepts the file name as a parameter. This makes the code unit-testable outside of Kubernetes.